### PR TITLE
fix planning_years to include pa_last_year_planned if DPs are tabled …

### PR DESCRIPTION
**Pull request by ktw1016**
…but DRRs not tabled

Check out if these look okay: `crso_by_prog_exp` and `dp_rev_split`

### CODE REVIEW

File: server/src/models/finances/constants.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
 import _ from "lodash";
 import {
   public_account_years,
-  planning_years,
+  correct_planning_years,
```
Comment (ktw1016) [inline server/src/models/finances/constants.js:4]: Nothing in this file is used anyways. Is this safe to delete?

File: server/src/models/constants.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const public_account_years = [
   "pa_last_year",
 ];
 
-export const planning_years = [
+export const correct_planning_years = [
```
Comment (ktw1016) [inline server/src/models/constants.js:11]: Same thing for this. It seems like this is not being used anywhere

File: server/src/models/constants.js [hunk 3]
```diff
@@ -8,12 +8,6 @@ export const public_account_years = [
   "pa_last_year",
 ];
 
-export const planning_years = [
-  "planning_year_1",
-  "planning_year_2",
-  "planning_year_3",
-];
-
```
Comment (ktw1016) [inline server/src/models/constants.js:16]: Not in use anywheres either

File: server/src/models/finances/constants.js (hunk 1 above)
Comment (Stephen-ONeil) [reply server/src/models/finances/constants.js:4]: I think that's more because the finance section of the API is old/unfinished, since we haven't decided to move to it yet. In theory this stuff would be used eventually, better to keep it around/up to date maybe.

File: client/src/panels/panel_declarations/finances/welcome_mat/welcome_mat_exp_program_spending.js [hunk 4]
```diff
@@ -134,7 +135,7 @@ export const format_and_get_exp_program_spending = (type, subject) => {
         return tick === _.first(plan_ticks) || tick === _.last(plan_ticks);
       } else {
         return (
-          tick === gap_year ||
+          tick === _.nth(all_ticks, _.round(all_ticks.length / 2)) ||
```
Comment (Stephen-ONeil) [inline client/src/panels/panel_declarations/finances/welcome_mat/welcome_mat_exp_program_spending.js:138]: This'll break if the number of history/planned years changes. Why the change?

Comment (ktw1016) [reply client/src/panels/panel_declarations/finances/welcome_mat/welcome_mat_exp_program_spending.js:138]: Because ticks look a bit bare empty with too few tick labels rendered. I'm simply rendering an additional tick label in the middle. How will this break?

**Comment by ktw1016** [comment]
Not all `planning_years` were replaced because `pa_last_year_planned` doesn't have _rev, _spa, _gross in the data (maybe we should pull that data too?)

so cols need to be defined separately from just `planning_years` and also for that reason, `dp_rev_split` panel should use `planning_years`

**Comment by Stephen-ONeil** [comment]
Hmm, that's annoying. Maybe `extended_planning_years` would be a good name then. `correct_planning_years` is off, because their correctness depends on context (correct for an FTE/spending panel, not correct for a planned revenue panel).

File: client/src/panels/panel_declarations/finances/welcome_mat/welcome_mat_exp_program_spending.js (hunk 4 above)
Comment (Stephen-ONeil) [reply client/src/panels/panel_declarations/finances/welcome_mat/welcome_mat_exp_program_spending.js:138]: Ah, I thought this was specifically trying to dynamically locate the gap year for some reason. So it's just making sure that the label for the middle tick always renders? Ok.
